src/librustc_middle/ty/layout.rs

```diff
@@ -2281,7 +2281,7 @@ impl<'tcx> ty::Instance<'tcx> {
     // or should go through `FnAbi` instead, to avoid losing any
     // adjustments `FnAbi::of_instance` might be performing.
     fn fn_sig_for_fn_abi(&self, tcx: TyCtxt<'tcx>) -> ty::PolyFnSig<'tcx> {
-        let ty = self.monomorphic_ty(tcx);
+        let ty = self.ty(tcx, ty::ParamEnv::reveal_all());
```

Comment:
Can you leave a FIXME for passing through a `ty::ParamEnv` here? (or maybe make the change and leave `rustc_codegen_{ssa,llvm}` passing `ty::ParamEnv::reveal_all()`, but I don't know how doable it is)

Reply:
Added a FIXME.